openstack/blockstorage/extensions/volumeactions/requests.go

```diff
@@ -241,6 +242,14 @@ type UploadImageOpts struct {
 
 	// Force image creation, usable if volume attached to instance.
 	Force bool `json:"force,omitempty"`
+
+	// Visibility defines who can see/use the image.
+	// supported since 3.1 microversion
+	Visibility images.ImageVisibility `json:"visibility,omitempty"`
```

Comment:
For posterity:

This field was added with a microversion, but Cinder v3 support was merged using microversion 3.27 as the "base". So I consider anything prior to that to be grandfathered, such as this field.

Otherwise, this field would be a pointer, as recently changed here: https://github.com/gophercloud/gophercloud/pull/1854

Reply:
Are you sure about this? According to the latest code it is still checks for the explicit 3.1 microversion: https://github.com/openstack/cinder/blob/e921b3ae6c462f5f844485f7da7164e194efd00c/cinder/api/contrib/volume_actions.py#L232..L240

Reply:
Sorry - to clarify: Cinder v3 support was merged into Gophercloud using microversion 3.27 as the "base".